sql/analyzer/resolve_subqueries.go

```diff
@@ -289,6 +289,10 @@ func nodeIsCacheable(n sql.Node, lowestAllowedIdx int) bool {
 				}
 			}
 		} else if sqa, ok := node.(*plan.SubqueryAlias); ok {
+			if sqa.CTESource {
```

Comment:
Test?

Reply:
this one is pretty easy to test, added

Reply:
nvm i can't replicate this in old version, we don't have the info to know we are nested in an active subquery expression while converting the CTE